lib/tempfile.rb

```diff
@@ -87,50 +87,64 @@
 # same Tempfile object from multiple threads then you should protect it with a
 # mutex.
 class Tempfile < DelegateClass(File)
-  # Creates a temporary file with permissions 0600 (= only readable and
-  # writable by the owner) and opens it with mode "w+".
+
+  # Creates a file in the underlying file system;
+  # returns a new \Tempfile object base on that file.
+  #
+  # If possible, consider instead using Tempfile.create, which:
+  #
+  # - Avoids the performance cost of delegation,
+  #   incurred when Tempfile.new calls its superclass <tt>DelegateClass(File)</tt>.
+  # - Does not rely on a finalizer to close and unlink the file,
+  #   which can be unreliable.
+  #
+  # Creates and returns file whose:
+  #
+  # - Class is \Tempfile (not \File, as in Tempfile.create).
+  # - Directory is the system temporary directory (system-dependent).
+  # - Generated filename is unique in that directory.
+  # - Permissions are <tt>0600</tt>;
+  #   see {File Permissions}[https://docs.ruby-lang.org/en/master/File.html#label-File+Permissions].
+  # - Mode is <tt>'w+'</tt> (read/write mode, positioned at the end).
+  #
+  # The file is not removed automatically,
+  # and so should be explicitly removed.
```

Comment:
I'm not sure if this is true. There's a finalizer defined to remove the tempfile when this object is swept by the garbage collector. E.g.

```ruby
irb(main):010:0> f = Tempfile.new
=> #<File:/var/folders/bf/7gzcjlld0ynbrnj0clyc_jgh0000gn/T/20220506-63138-9qcomk>
irb(main):011:0> path = f.path
=> "/var/folders/bf/7gzcjlld0ynbrnj0clyc_jgh0000gn/T/20220506-63138-9qcomk"
irb(main):012:0> File.exist?(path)
=> true
irb(main):013:0> f = nil
=> nil
irb(main):014:0> GC.start
=> nil
irb(main):015:0> File.exist?(path)
=> false
```

Reply:
That's sort of what the previous doc said, I think.  What should we say here?

Reply:
I don't think it said anything about this for the documentation for `Tempfile.new`.

It's certainly true for `Tempfile.create`:

> If no block is given, ... the created file is not removed automatically.

I think we can say something like:

```
The file is removed when the object has died and reclaimed by the garbage collector.
```

Reply:
Fixed.